examples/1_structure.cpp

```diff
@@ -47,5 +47,10 @@ int main()
     Dataset dataset = Dataset(Datatype::DOUBLE, Extent{1});
     mass_scalar.resetDataset(dataset);
 
+    /* Required Records and RecordComponents are created automatically.
+     * Initialization has to be done explicitly by the user. */
+    electrons["position"][RecordComponent::SCALAR].resetDataset(dataset);
```

Comment:
I just realized `resetDataset` in `(Patch)RecordComponent` lacks doxygen docs. What does it do, overwrite a dataset?

Reply:
It overwrites the default Dataset (single ```char```) with a user-defined dataset, including datatype, extent, chunking, compression, etc.
The Dataset is used when flushing a RecordComponent to disk, see [```RecordComponent::flush()```](https://github.com/openPMD/openPMD-api/blob/7501e3435913216812f3efe8707b30892b3aa915/src/RecordComponent.cpp#L97-L106) for example.

Reply:
At this point I am not sure a default *valid* dataset is the best solution. It surely is helpful for prototyping, but having a ```flush()``` fail with an uninitialized, default dataset sends a clear message.